tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/graphql/resolvers/TaskVariablesFetcher.java

```diff
@@ -14,55 +14,35 @@
  * NOTHING IN THIS AGREEMENT EXCLUDES OR RESTRICTS A PARTY’S LIABILITY FOR (A) DEATH OR PERSONAL INJURY CAUSED BY THAT PARTY’S NEGLIGENCE, (B) FRAUD, OR (C) ANY OTHER LIABILITY TO THE EXTENT THAT IT CANNOT BE LAWFULLY EXCLUDED OR RESTRICTED.
  */
-package io.camunda.tasklist.webapp.graphql.query;
+package io.camunda.tasklist.webapp.graphql.resolvers;
 
 import static io.camunda.tasklist.webapp.graphql.TasklistGraphQLContextBuilder.VARIABLE_DATA_LOADER;
 
-import graphql.kickstart.tools.GraphQLResolver;
+import graphql.schema.DataFetcher;
 import graphql.schema.DataFetchingEnvironment;
 import graphql.schema.SelectedField;
 import io.camunda.tasklist.store.VariableStore;
+import io.camunda.tasklist.store.VariableStore.GetVariablesRequest;
 import io.camunda.tasklist.webapp.graphql.entity.TaskDTO;
 import io.camunda.tasklist.webapp.graphql.entity.VariableDTO;
-import io.camunda.tasklist.webapp.mapper.TaskMapper;
 import java.util.List;
-import java.util.Set;
 import java.util.concurrent.CompletableFuture;
 import java.util.stream.Collectors;
 import org.dataloader.DataLoader;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Component;
 
-@Component
-public class TaskResolver implements GraphQLResolver<TaskDTO> {
+public class TaskVariablesFetcher implements DataFetcher<CompletableFuture<List<VariableDTO>>> {
 
-  @Autowired private TaskMapper taskMapper;
-
-  public CompletableFuture<List<VariableDTO>> getVariables(
-      TaskDTO task, DataFetchingEnvironment dfe) {
-    final DataLoader<VariableStore.GetVariablesRequest, List<VariableDTO>> dataloader =
-        dfe.getDataLoader(VARIABLE_DATA_LOADER);
+  @Override
+  public CompletableFuture<List<VariableDTO>> get(final DataFetchingEnvironment env)
+      throws Exception {
+    final DataLoader<GetVariablesRequest, List<VariableDTO>> dataloader =
+        env.getDataLoader(VARIABLE_DATA_LOADER);
 
     return dataloader.load(
         VariableStore.GetVariablesRequest.createFrom(
-            TaskDTO.toTaskEntity(task), getFieldNames(dfe)));
-  }
-
-  public String getProcessName(TaskDTO task) {
-    return taskMapper.getProcessName(task);
-  }
-
-  public String getName(TaskDTO task) {
-    return taskMapper.getName(task);
-  }
-
-  public String getTaskDefinitionId(TaskDTO task) {
-    return task.getFlowNodeBpmnId();
-  }
-
-  private static Set<String> getFieldNames(DataFetchingEnvironment dataFetchingEnvironment) {
-    return dataFetchingEnvironment.getSelectionSet().getFields().stream()
-        .map(SelectedField::getName)
-        .collect(Collectors.toSet());
+            TaskDTO.toTaskEntity(env.getSource()),
+            env.getSelectionSet().getFields().stream()
```

Comment:
As we are using completable future on data fetching, is it possible have a dat inconsistency here? 

Should we place this on a thread-safe operation?

Reply:
this is equivalent to what we had before https://github.com/camunda/zeebe/blob/5be39bf535648c7ad23067c2bc6f8b9b038e2c54/tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/graphql/query/TaskResolver.java#L41-L49

it avoid N+1 query problem, and groups the querying of tasks variables in a single method invocation